gslib/utils/user_agent_helper.py

```diff
@@ -32,11 +33,28 @@ def GetUserAgent(args, metrics_off=True):
   """
   user_agent = ' gsutil/%s' % gslib.VERSION
   user_agent += ' (%s)' % sys.platform
-  user_agent += ' analytics/%s ' % ('disabled' if metrics_off else 'enabled')
+  user_agent += ' analytics/%s' % ('disabled' if metrics_off else 'enabled')
+  user_agent += ' interactive/%s' % system_util.IsRunningInteractively()
+
+  if len(args) > 0:
+    user_agent += ' command/%s' % args[0]
+
+    if args[0] in ['cp', 'mv', 'rsync']:
+      # Any cp, mv or rsync commands that use daisy chain mode should be noted
+      # as that represents a unique use case that may be better served by the
+      # storage transfer service.
+      try:
+        src = StorageUrlFromString(args[-2])
```

Comment:
Potential IndexError. Maybe you will need a test case for ['cp'] like you have one for help

Reply:
Good call!